tests/integration/telemetry/stackdriver/testdata/security_authz_audit/server_audit_all_log.json.tmpl

```diff
@@ -0,0 +1,25 @@
+{
```

Comment:
How can I tell if this is a audit log entry or normal access log? I think we'd need to verify the name the log entry?

Reply:
audit log have less labels than access log. Now audit log do not support add a name for it.

Reply:
No audit log is named differently. See https://github.com/istio/proxy/blob/45ef14104ad84554f3600e5340dd59d9764a043c/extensions/stackdriver/log/logger_test.cc#L138 for an example full audit log message, which is named as `projects/test_project/logs/server-istio-audit-log`, whereas normal access log is named as `server-accesslog-stackdriver`.

I think what you should do is 1) extend [fake stackdriver](https://github.com/istio/proxy/blob/45ef14104ad84554f3600e5340dd59d9764a043c/test/envoye2e/stackdriver_plugin/fake_stackdriver.go#L141) to support distinguish different log types and returns audit log entries on a different REST URL (currently all log entries are returned by [/logentries](https://github.com/istio/proxy/blob/45ef14104ad84554f3600e5340dd59d9764a043c/test/envoye2e/stackdriver_plugin/fake_stackdriver.go#L486). we should create a new url like `/auditlogentries`) 2) then update the test here to update the listLogEntry method to list audit log entry only.

Reply:
Yeah thanks for showing also this great approach and elaborating on the details. Since my approach is also workable and trying to catch the 1.8 release. Maybe we can merge this one and then refactor code shortly.

Reply:
We can also add logname from WriteLogEntriesRequest to logEntry and then filter here in the test.. that might be easiest.. @bianpengyuan : wdyt? 

Definitely, something to follow-up quickly if this PR needs to be submitted sooner..